components/telemetry-controller/controllers/loggingconfiguration_controller.go

```diff
@@ -0,0 +1,318 @@
+Copyright 2021.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"bytes"
+	"context"
+
+	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	telemetryv1alpha1 "github.com/kyma-project/kyma/components/telemetry-controller/api/v1alpha1"
+)
+
+const (
+	configMapFinalizer   = "FLUENT_BIT_CONFIG_MAP"
+	environmentFinalizer = "FLUENT_BIT_ENV"
+	filesFinalizer       = "FLUENT_BIT_FILES"
+)
+
+// LoggingConfigurationReconciler reconciles a LoggingConfiguration object
+type LoggingConfigurationReconciler struct {
+	client.Client
+	Scheme                  *runtime.Scheme
+	FluentBitConfigMap      types.NamespacedName
+	FluentBitDaemonSet      types.NamespacedName
+	FluentBitEnvSecret      types.NamespacedName
+	FluentBitFilesConfigMap types.NamespacedName
+}
+
+//+kubebuilder:rbac:groups=telemetry.kyma-project.io,resources=loggingconfigurations,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=telemetry.kyma-project.io,resources=loggingconfigurations/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=telemetry.kyma-project.io,resources=loggingconfigurations/finalizers,verbs=update
+//+kubebuilder:rbac:groups="",resources=configmaps,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups="",resources=secrets,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups="",resources=pods,verbs=delete;list;watch
+//+kubebuilder:rbac:groups="apps",resources=daemonsets,verbs=get;list;watch
+
+// Reconcile is part of the main kubernetes reconciliation loop which aims to
+// move the current state of the cluster closer to the desired state.
+//
+// For more details, check Reconcile and its Result here:
+// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.8.3/pkg/reconcile
+func (r *LoggingConfigurationReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	log := log.FromContext(ctx)
+
+	var config telemetryv1alpha1.LoggingConfiguration
+	if err := r.Get(ctx, req.NamespacedName, &config); err != nil {
+		log.Error(err, "unable to fetch LoggingConfiguration")
```

Comment:
Currently, if a CR is getting deleted, this error is getting logged (with a stack trace, etc.). It's not really nice since deleting a CR should not be treated as an error. Could we have a check or something similar to this: https://github.com/kyma-project/kyma/blob/main/components/application-operator/pkg/application-controller/reconciler.go#L74 (maybe also with an additional log entry for a deleted CR)?

It would be cool to have something like an audit log:

```
Added pipeline-a
Added pipeline-b
Pipeline-a deleted
```

Reply:
Tried to address this suggestion. However, it is not always easy to figure out if a resource has been created or updated.